executor/linux/step.go

```diff
@@ -214,6 +215,13 @@ func (c *client) StreamStep(ctx context.Context, ctn *pipeline.Container) error
 		return err
 	}
 
+	secretValues := []string{}
+
+	for _, secret := range ctn.Secrets {
+		s := ctn.Environment[strings.ToUpper(secret.Target)]
+		secretValues = append(secretValues, s)
+	}
```

Comment:
Does this implementation account for external `secrets`?

https://go-vela.github.io/docs/tour/secrets/

https://go-vela.github.io/docs/usage/examples/secrets_external/

https://go-vela.github.io/docs/reference/yaml/secrets/#the-origin-tag

These secrets are unique since they are fetched by a Docker container and written to the build workspace.

AFAICT, it appears this checks for internal `secrets` declared in the pipeline for the `step`:

https://go-vela.github.io/docs/usage/examples/secrets_internal/

If possible, I think we should try to account for external secrets declared in the pipeline as well.

Reply:
that's a good point, too. they should all be in `/vela/secrets/` (at least for how it currently works).

Reply:
I'm not sure we will be able to catch external secrets because the vela worker code won't be running in the workspace that the `/vela/secrets` would exist within. 

It's possible you could make calls to the Docker volume to get them but I'm not sure we have those functions within the runtime.

Reply:
To elaborate (kinda just word vomited), Vela external secrets are entirely controller by plugins meaning the worker's responsibility, in that case, is just managing the container, and the secrets are written to the dynamic volume that is created. 

So, if the worker wanted to see what those values are it would need to interrogate the volume that was spun up for that particular build and then get the values and do its sanitization. I'm not saying it's impossible but I don't think we are set up runtime-wise yet to make calls to the volumes to analyze content within them.

Reply:
With that being said, should this PR be only internal secrets? Create a separate issue for external secret masking?

Reply:
I would vote scope to internal but we should hear from others. I think external secrets need to be re-evaluated again. 😢 

We know a lot more now than when we did the first couple of attempts at writing the feature and I think it needs to be adapted.  I've seen many requests to make them ENV while I'm not sure that's entirely possible we could look into it again. That would potentially give us more options for sanitization but I don't know at this point.

Reply:
Yeah, I wasn't sure if it was possible or not 🤷 

Was thinking we could try and add code that inspects the build workspace (`/vela`) to get the secret values?

If everyone is on board for keeping this feature scoped to `internal` secrets I think I can get behind that 😃 

We can always worry about this functionality later for `external` secrets

Could I request a story is written for it though?

i.e. mask `external` secret in the logs

Reply:
[External Secrets Masking Issue](https://github.com/go-vela/community/issues/465)